app/controllers/application_controller.rb

```diff
@@ -507,6 +507,15 @@ def check_permission_to_wrangle
     end
   end
 
+  # Checks if user is allowed to see related page if parent item is hidden or in unrevealed collection
+  def check_parent_visibility(parent)
```

Comment:
Should this be just "check_visibility"? Otherwise, I would expect this to call a variant of `#parent` on the object passed here

Reply:
There's no unified way to get the parent for all these. It's not named `check_visibility` because that's already a thing, it's two methods above this one. But I don't like this name either.

Reply:
Maybe either `check_visibility_of` or `check_visibility_for`? The other one is generically for the current page, where this is for the passed in object/record